src/iris/api.py

```diff
@@ -4002,67 +4003,43 @@ def on_get(self, req, resp):
-                                                                  WHERE `created` > (CURRENT_DATE - INTERVAL 29 DAY)
-                                                                  AND `created` < (CURRENT_DATE - INTERVAL 1 DAY)
-                                                                  AND `active` = FALSE
-                                                                  AND NOT ISNULL(`owner_id`)
-                                                                  AND NOT ISNULL(`updated`)
-                                                                  ORDER BY time_to_claim) as time_to_claim
-                                                            WHERE (SELECT @row_id := @row_id + 1)
-                                                            BETWEEN @incident_count/2.0 AND @incident_count/2.0 + 1)''',
-            'total_applications': 'SELECT COUNT(*) FROM `application` WHERE `auth_only` = FALSE'
-        }
+    def __init__(self, config):
+        cfg = config.get('app-stats', {})
+        # Calculate stats in real time (True), or query offline stats
+        # generated by the app stats daemon (False)
+        self.real_time = cfg.get('real_time', True)
 
-        stats = {}
+    def on_get(self, req, resp):
 
         fields_filter = req.get_param_as_list('fields')
-        fields = queries.viewkeys()
-        if fields_filter:
-            fields &= set(fields_filter)
-
         connection = db.engine.raw_connection()
         cursor = connection.cursor()
-        for key in fields:
-            start = time.time()
-            cursor.execute(queries[key])
-            result = cursor.fetchone()
-            if result:
-                result = result[-1]
-            else:
-                result = None
-            logger.info('Stats query %s took %s seconds', key, round(time.time() - start, 2))
-            stats[key] = result
+        if self.real_time:
+            stats = app_stats.calculate_gobal_stats(connection, cursor, fields_filter=fields_filter)
+
+        else:
+            # select the most current global stats entry
+            query = '''
+                SELECT `median_seconds_to_claim_last_month`, `total_incidents_today`, `total_active_users`, `total_plans`, `total_messages_sent`,
+                `total_applications`, `pct_incidents_claimed_last_month`, `total_incidents`, `total_messages_sent_today`, `timestamp`
+                FROM `global_stats` gs1
+                WHERE `timestamp` = (SELECT MAX(`timestamp`) from `global_stats` gs2 WHERE gs2.`timestamp` = gs1.`timestamp`)
+            '''
+            cursor.execute(query)
+
+            if cursor.rowcount == 0:
+                logger.error('Error retrieving global stats from db')
+                raise HTTPInternalServerError('Error retrieving global stats from db')
+
+            # build dictionary of stat names and values
+            desc = cursor.description
```

Comment:
If you do `cursor = connection.cursor(db.dict_cursor)`, the rows cursor.fetchall() returns will be a dict mapping column names to values, which alleviates having to construct that dict manually here.

But then, that likely breaks the rest of the uses of this cursor as those all expect to not be using a dict cursor.

As you already know the fields which will be returned, as you listed them explicitly in your query, you could avoid relying on this metadata and construct the dict manually. eg:

```
columns = ('median_seconds_to_claim_last_month', 'total_incidents_today', 'total_active_users', 'total_plans', 'total_messages_sent', 'total_applications', 'pct_incidents_claimed_last_month', 'total_incidents', 'total_messages_sent_today', 'timestamp')
row = cursor.fetchone()
stats = {column: row[i] for column, i in enumerate(columns)}
```